src/TriggerBinding/SqlTriggerListener.cs

```diff
@@ -391,19 +394,33 @@ private async Task<long> InsertGlobalStateTableRowAsync(SqlConnection connection
 
             string insertRowGlobalStateTableQuery = $@"
                 {AppLockStatements}
-
+                -- For back compatibility copy the lastSyncVersion from _oldUserFunctionId if it exists.
                 IF NOT EXISTS (
                     SELECT * FROM {GlobalStateTableName}
                     WHERE UserFunctionID = '{this._userFunctionId}' AND UserTableID = {userTableId}
                 )
+                BEGIN
+                    -- Migrate LastSyncVersion from oldUserFunctionId if it exists and delete the record
+                    Declare @lastSyncVersion bigint;
+                    Select @lastSyncVersion = LastSyncVersion from az_func.GlobalState where UserFunctionID = '{this._oldUserFunctionId}' AND UserTableID = {userTableId}
+                    IF @lastSyncVersion is NULL
+                        SET @lastSyncVersion = {(long)minValidVersion};
+                    ELSE
+                        Delete from az_func.GlobalState where UserFunctionID = '{this._oldUserFunctionId}' AND UserTableID = {userTableId}
+                    
                     INSERT INTO {GlobalStateTableName}
                     VALUES ('{this._userFunctionId}', {userTableId}, {(long)minValidVersion}, GETUTCDATE());
```

Comment:
Shouldn't this be using @lastSyncVersion? That'll be the old value is we're migrating over.